subprojects/base-services/src/test/groovy/org/gradle/internal/time/DefaultTimerTest.groovy

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.gradle.internal.time
+
+import spock.lang.Specification
+
+import java.util.concurrent.TimeUnit
+
+class DefaultTimerTest extends Specification {
+
+    private static final long TEST_BASE_TIME = 641353121231L;
+
+    private TimeSource timeProvider = Mock(TimeSource);
+    private DefaultTimer clock;
+
+    void setup() {
+        1 * timeProvider.nanoTime() >> TEST_BASE_TIME;
+        clock = new DefaultTimer(timeProvider);
+    }
+
+    def testOnlySecondsTwoDigits() throws Exception {
+        when:
+        setDtMs(51243);
+
+        then:
+        clock.getElapsed() == "51.243 secs"
+    }
+
+    def testOnlySecondsEvenMs() {
+        when:
+        setDtMs(4000);
+
+        then:
+        clock.getElapsed() == "4.0 secs"
+    }
+
+    def testMinutesAndSeconds() {
```

Comment:
This is an existing test, renamed. I'm not sure it's worth refactoring at this time.